x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/optimizer/LogicalPlanOptimizer.java

```diff
@@ -612,6 +622,65 @@ protected Expression maybeSimplifyNegatable(Expression e) {
 
     }
 
+    /**
+     * Look for numerical literals on the right hand side of a {@link BinaryComparison}s that are out of range for the left hand side data
+     * type and replace the comparison by a literal TRUE or FALSE. This rule should be placed after {@link LiteralsOnTheRight}.
+     */
+    private static class OutOfRangeBinaryComparison extends OptimizerRules.OptimizerExpressionRule<BinaryComparison> {
+
+        OutOfRangeBinaryComparison() {
+            super(TransformDirection.DOWN);
+        }
+
+        @Override
+        protected Expression rule(BinaryComparison bc) {
+            // TODO: add tests with null literals if they do not already exist
+            if (bc.left().dataType().isNumeric() == false || bc.right().dataType().isNumeric() == false) {
+                return bc;
+            }
+
+            if (bc.right() instanceof Literal r) {
+                if (r.value() == null || isInRange(bc.left().dataType(), r.value())) {
+                    return bc;
+                }
+
+                if (bc instanceof org.elasticsearch.xpack.ql.expression.predicate.operator.comparison.Equals || bc instanceof NullEquals) {
+                    return Literal.falseWithSource(bc.source());
+                }
+                if (bc instanceof NotEquals) {
+                    return Literal.trueWithSource(bc.source());
+                }
+
+                boolean isPositive = OptimizerUtils.isPositive(r);
+                if (bc instanceof GreaterThan || bc instanceof GreaterThanOrEqual) {
+                    return isPositive ? Literal.falseWithSource(bc.source()) : Literal.trueWithSource(bc.source());
+                }
+                if (bc instanceof LessThan || bc instanceof LessThanOrEqual) {
+                    return isPositive == false ? Literal.falseWithSource(bc.source()) : Literal.trueWithSource(bc.source());
+                }
```

Comment:
In all cases the conditions create a true/false so do that and then wrap the result in a literal
```
boolean result = false;
if (bc instanceof Equals || NullEquals) {
 value = true;
}
else  if (bc instanceof Equals)
...

return new Literal.of(value, bc);
```